_posts/2019-08-20-service-objects.markdown

```diff
@@ -0,0 +1,100 @@
+---
+layout: post
+title:  "Service Objects: beyond fat models and skinny controllers"
+date: 2019-09-03 10:00:00
+categories: ["rails"]
+author: "cleiviane"
+---
+
+Service Objects are a controversial idea for several different reasons: some developers like to use them, others like to use similar patterns, and some think that they are just unnecessary because they prefer fat models.  
+
+Here at Ombu Labs we like to use service objects whenever we can, we think it's a great way to keep our controllers skinny.
+
+In this post I would like to discuss my idea about service objects and why it's adopted by our team.
+
+<!--more-->
+
+## What is a Service Object?
+
+A Service is a stateless object that encapsulates a set of steps and usually has a single purpose. It's a great resource to decompose fat Active Record models while still maintaining the controllers thin, because we take away some validations which don't belong to a model nor a controller.  
```

Comment:
```suggestion
A Service is a stateless object that encapsulates a set of steps and usually has a single purpose. It's a great resource to decompose fat Active Record models while still keeping the controllers thin, because we take away some validations which don't belong to a model nor a controller.  
```